hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java

```diff
@@ -1103,11 +1104,24 @@ public List<HddsProtos.DatanodeUsageInfoProto> getDatanodeUsageInfo(
 
     // return count number of DatanodeUsageInfoProto
     return datanodeUsageInfoList.stream()
-        .map(each -> each.toProto(clientVersion))
+        .map(each -> each.toProto(clientVersion,
+                getContainerCount(each.getDatanodeDetails())))
         .limit(count)
         .collect(Collectors.toList());
   }
 
+  private long getContainerCount(DatanodeDetails datanodeDetails) {
+    long containerCount = -1;
+    try {
+      Set<ContainerID> containers = scm.getScmNodeManager()
+              .getContainers(datanodeDetails);
+      containerCount = containers.size();
+    } catch (NodeNotFoundException e) {
+      LOG.error("Node not found, error: ", e);
+    }
+    return containerCount;
+  }
```

Comment:
Countainer count should be set on `DatanodeUsageInfo` by `SCMNodeManager`.  It does not belong to the protocol server.

Reply:
@adoroszlai Done. Bu maybe we could go even one more step further. What about adding containerCount into DatanodeDetails, or even adding a Set<ContainerID> into DatanodeDetails?

Reply:
> What about adding containerCount into DatanodeDetails, or even better... we could add a Set of ContainerID into DatanodeDetails?

I don't think we should add either of those extra pieces of info in `DatanodeDetails`.  It contains static information, while container count or container IDs are dynamically changing.  The latter could also grow too large.